lib/Jsrt/Jsrt.cpp

```diff
@@ -2746,3 +3230,504 @@ CHAKRA_API JsRunSerializedScriptWithCallback(_In_ JsSerializedScriptLoadSourceCa
+
+            threadContext->TTDLog->UpdateInflateMapForFreshScriptContexts();
+
+            HostScriptContextCallbackFunctor callbackFunctor(context, &JsrtContext::OnScriptLoad_TTDCallback);
+            threadContext->BeginCtxTimeTravel(context->GetScriptContext(), callbackFunctor);
+            context->GetScriptContext()->GetDebugContext()->SetDebuggerMode(Js::DebuggerMode::Debugging);
+
+            //initialize the core image but we need to disable debugging while this happens
+            threadContext->TTDLog->PushMode(TTD::TTDMode::ExcludedExecution);
+            context->GetScriptContext()->InitializeCoreImage_TTD();
+            threadContext->TTDLog->PopMode(TTD::TTDMode::ExcludedExecution);
+
+            context->GetScriptContext()->InitializeDebuggingActionsAsNeeded_TTD();
+
+            scriptContext->InitializeDebugging();
+            context->GetScriptContext()->GetDebugContext()->GetProbeContainer()->InitializeInlineBreakEngine(runtime->GetJsrtDebugManager());
+            context->GetScriptContext()->GetDebugContext()->GetProbeContainer()->InitializeDebuggerScriptOptionCallback(runtime->GetJsrtDebugManager());
+            threadContext->GetDebugManager()->SetLocalsDisplayFlags(Js::DebugManager::LocalsDisplayFlags::LocalsDisplayFlags_NoGroupMethods);
+        }
+        catch(...)
+        {
+            //something went horribly wrong
+
+            *targetStartSnapTime = -1;
+        }
+    }
+
+    return (*targetStartSnapTime != -1) ? JsNoError : JsErrorFatal;
+#endif
+}
+
+CHAKRA_API JsTTDMoveToTopLevelEvent(_In_ INT64 snapshotTime, _In_ INT64 eventTime)
+{
+#if !ENABLE_TTD_DEBUGGING
+    return JsErrorCategoryUsage;
+#else
+    JsrtContext *currentContext = JsrtContext::GetCurrent();
+    Js::ScriptContext* scriptContext = currentContext->GetScriptContext();
+    if(scriptContext->GetThreadContext()->TTDLog == nullptr)
+    {
+        AssertMsg(false, "Should only happen in TT debugging mode.");
+        return JsErrorCategoryUsage;
+    }
+
+    TTD::EventLog* elog = scriptContext->GetThreadContext()->TTDLog;
+    JsErrorCode res = JsNoError;
+
+    //a special indicator to use the time from the argument flag
+    if(eventTime == -2)
+    {
+        eventTime = scriptContext->GetThreadContext()->TTDLog->GetKthEventTime(Js::Configuration::Global.flags.TTDStartEvent);
+    }
+
+    try
+    {
+        elog->PushMode(TTD::TTDMode::ExcludedExecution);
+        BEGIN_JS_RUNTIME_CALLROOT_EX(scriptContext, false)
+        {
+            elog->DoSnapshotInflate(snapshotTime);
+        }
+        END_JS_RUNTIME_CALL(scriptContext);
+        elog->PopMode(TTD::TTDMode::ExcludedExecution);
+
+        //If the log has a BP requested then we should set the actual bp here
+        if(scriptContext->GetThreadContext()->TTDLog->HasPendingTTDBP())
+        {
+            TTD::TTDebuggerSourceLocation bpLocation;
+            scriptContext->GetThreadContext()->TTDLog->GetPendingTTDBPInfo(bpLocation);
```

Comment:
This should happen in the JsRTDiag code. We need to finish merging in some of my debugger code there to full light up this scenario but currently blocked on there.